scipy/sparse/dia.py

```diff
@@ -237,6 +237,40 @@ def sum(self, axis=None, dtype=None, out=None):
 
     sum.__doc__ = spmatrix.sum.__doc__
 
+    def _add_sparse(self, other):
+        """If other is also of type dia_matrix,
+        add all the diagonals, and return a result of type dia_matrix.
+        Otherwise, convert to csr_matrix, which is the default behaviour
+        of the _add_sparse function of parent class spmatrix.
+        """
+
+        # Check if other is also of type dia_matrix
+        if not isinstance(other, type(self)):
+            # If other is not of type dia_matrix, default to
+            # converting to csr_matrix, as is done in the _add_sparse
+            # method of parent class spmatrix
+            return self.tocsr()._add_sparse(other)
+        else:
```

Comment:
Since the if-case returns, I'd drop the `else` here and avoid indenting the rest of the function.

Reply:
ok